tests/TestCase/Event/EventManagerTest.php

```diff
@@ -152,6 +152,67 @@ public function testAttachMultipleEventKeys()
     }
 
     /**
+     * Tests attached event listeners for matching key pattern
+     *
+     * @return void
+     */
+    public function testMatchingListeners()
+    {
+        $manager = new EventManager();
+        $manager->attach('fakeFunction1', 'fake.event');
+        $manager->attach('fakeFunction2', 'real.event');
+        $manager->attach('fakeFunction3', 'test.event');
+        $manager->attach('fakeFunction4', 'event.test');
+
+        $this->assertArrayHasKey('fake.event', $manager->matchingListeners('fake'));
+        $this->assertArrayHasKey('real.event', $manager->matchingListeners('real'));
+        $this->assertArrayHasKey('test.event', $manager->matchingListeners('test'));
+        $this->assertArrayHasKey('event.test', $manager->matchingListeners('test'));
+        $this->assertArrayHasKey('test.event', $manager->matchingListeners('^test'));
+        $this->assertArrayHasKey('event.test', $manager->matchingListeners('test$'));
```

Comment:
Exposing regular expressions as an API interface is powerful but also more complicated.